app/apollo/resolvers/channel.js

```diff
@@ -330,37 +325,27 @@ const channelResolvers = {
             // Note: if failure occurs after this point, the data may already have been stored by storageFactory even if the Version document doesnt get saved
 
             // Save Version
-            const dObj = await models.DeployableVersion.create( versionObj );
+            await models.DeployableVersion.create( versionObj );
 
             // Keep version uuid for later use when creating subscriptions
             v.uuid = versionObj.uuid;
-
-            // Attempt to update Version references the channel (the duplication is unfortunate and should be eliminated in the future)
-            try {
-              const channelVersionObj = {
-                uuid: versionObj.uuid,
-                name: versionObj.name,
-                description: versionObj.description,
-                created: dObj.created
-              };
-              await models.Channel.updateOne(
-                { org_id, uuid: newChannelObj.uuid },
-                { $push: { versions: channelVersionObj } }
-              );
-            } catch(err) {
-              logger.error(err, `${queryName} failed to update the channel to reference the new Version '${versionObj.name}' / '${newChannelObj.uuid}' when serving ${req_id}.`);
-              // Cannot fail here, the Version has already been created.  Continue.
-            }
           }
           catch( e ) {
-            logger.error(e, `${queryName} failed to create version '${versionObj.name}' when serving ${req_id}.`);
+            logger.error(e, `${queryName} error creating version '${versionObj.name}' when serving ${req_id}.`);
             // Cannot fail here, the Channel has already been created.  Continue.
           }
         } ) );
 
         // Attempt to create subscription(s)
         await Promise.all( subscriptions.map( async (s) => {
           const version = versions.find( v => v.name === s.versionName );
+          if( !version ) {
+            logger.error({ req_id, user: whoIs(me), org_id, name }, `${queryName} unable to create subscription '${s.name}' when serving ${req_id}, version '${s.versionName}' was not created.`);
+          }
+          else if( !version.uuid ) {
+            logger.error({ req_id, user: whoIs(me), org_id, name }, `${queryName} unable to create subscription '${s.name}' when serving ${req_id}, version '${s.versionName}' failed creation.`);
+          }
+
```

Comment:
```suggestion
          if( !version || !version.uuid) {
            logger.error({ req_id, user: whoIs(me), org_id, name }, `${queryName} unable to create subscription '${s.name}' when serving ${req_id}, version '${s.versionName}' was not created.`);
          }
```

Reply:
There is a difference between the two states:
- `!version` means the caller of the API did not specify the version so it was not created
- `!version.uuid` means the caller of the API provided the version details but an error occurred when trying to create it

Keeping the messages (slightly) distinct can help explain behavior if/when either is encountered.